chainer/variable.py

```diff
@@ -52,6 +52,52 @@ def make_message(message):
         raise ValueError(make_message(msg))
 
 
+def variable_repr(var):
+    """Return the string representation of a variable.
+
+    Args:
+        var (~chainer.Variable): Input Variable.
+    .. seealso:: numpy.array_repr
+    """
+    xp = cuda.get_array_module(var)
+    if xp is numpy:
+        arr = var.data
+    else:
+        arr = var.data.get()
+
+    if arr.size > 0 or arr.shape == (0,):
+        if var.name:
+            prefix = 'variable ' + var.name + '('
+        else:
+            prefix = 'variable('
+        lst = numpy.array2string(arr, None, None, None, ', ', prefix)
+    else:  # show zero-length shape unless it is (0,)
+        lst = '[], shape=%s' % (repr(arr.shape),)
+        if var.name:
+            lst = var.name + lst
+    return '%s(%s)' % (prefix[:-1], lst)
```

Comment:
`prefix` will be undefined if `shape == (0,0)` etc.
I think how `prefix` is determined should be independent from `shape`.

Also, the operation `prefix[:-1]` seems redundant for me. 

In summary, how about the following logic?

```
    if var.name:
        prefix = 'variable ' + var.name
    else:
        prefix = 'variable'

    if arr.size > 0 or arr.shape == (0,):
        lst = numpy.array2string(arr, None, None, None, ', ', prefix + '(')
    else:  # show zero-length shape unless it is (0,)
        lst = '[], shape=%s' % (repr(arr.shape),)

    return '%s(%s)' % (prefix, lst)
```